### PR TITLE
test: Creating multi db rule to run tests multiple times against different named databases.

### PR DESCRIPTION
**Pull request by jainsahab**
Every test class needs to use it in the following way

```java

@ClassRule public static MultiDbRule multiDbRule = new MultiDbRule();

@BeforeClass
public static void beforeClass() throws Exception {
  DATASTORE = multiDbRule.getDatastore();
}
```

### CODE REVIEW

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/MultiDbRule.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        try {
+          // running with default Datastore
+          logger.log(Level.INFO, "Current database: default");
+          base.evaluate();
+
+          // running with test-db Datastore
+          logger.log(Level.INFO, "Current database: test-db");
```
Comment (kolea2) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/MultiDbRule.java:67]: nit - I would replace this with the CUSTOM_DB_ID variable

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreAggregationsTest.java [hunk 2]
```diff
@@ -89,7 +88,6 @@ public void testSumAggregation() {
         Query.newAggregationQueryBuilder()
             .over(baseQuery)
             .addAggregations(sum("marks").as("total_marks"))
-            .setNamespace(OPTIONS.getNamespace())
```
Comment (kolea2) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreAggregationsTest.java:92]: why remove the namespace?

**Review by kolea2 — COMMENTED** [review]
from the test logs:
```
[INFO] Running com.google.cloud.datastore.it.ITDatastoreTest
Dec 22, 2023 5:30:30 PM com.google.cloud.datastore.it.MultiDbRule$1 evaluate
INFO: Current database: test-db
[INFO] Tests run: 64, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 86.03 s -- in com.google.cloud.datastore.it.ITDatastoreTest
```

it doesn't look like the ITDatastoreTest ran for default - can you confirm?

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/MultiDbRule.java (hunk 1 above)
Comment (jainsahab) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/MultiDbRule.java]: fixed in 9c52cca.

**Comment by jainsahab** [comment]
> from the test logs:
> 
> ```
> [INFO] Running com.google.cloud.datastore.it.ITDatastoreTest
> Dec 22, 2023 5:30:30 PM com.google.cloud.datastore.it.MultiDbRule$1 evaluate
> INFO: Current database: test-db
> [INFO] Tests run: 64, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 86.03 s -- in com.google.cloud.datastore.it.ITDatastoreTest
> ```
> 
> it doesn't look like the ITDatastoreTest ran for default - can you confirm?

seems like logs are jumbled up for some reason, I have anyways included the `testName` as part of the log, so we can distinguish between them now.

```
Dec 28, 2023 2:56:00 PM com.google.cloud.datastore.it.MultiDbRule$1 evaluate
INFO: Running com.google.cloud.datastore.it.ITDatastoreAggregationsTest with database default
com.google.cloud.datastore.it.ITDatastoreAggregationsTest
Dec 28, 2023 2:56:03 PM com.google.cloud.datastore.it.MultiDbRule$1 evaluate
INFO: Running com.google.cloud.datastore.it.ITDatastoreAggregationsTest with database test-db
```

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreAggregationsTest.java (hunk 2 above)
Comment (kolea2) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreAggregationsTest.java:92]: ahh gotcha, thanks!

**Review by kolea2 — APPROVED** [review]

**Pull request merged**